docs/reference/modules/shard-recovery-desc.asciidoc

```diff
@@ -0,0 +1,16 @@
+Shard recovery is the process of initializing a shard copy, such as restoring a
+primary shard from a snapshot or creating a replica shard from a primary shard.
+When a shard recovery completes, the recovered shard is available for search
+and indexing.
+
+Recovery automatically occurs during the following processes:
+
+* When creating an index for the first time (`empty_store`).
+* Node startup or failure. This type of recovery is called a local store recovery (`existing_store`).
+* Creation of new replica shard copies from the primary (`peer`).
+* Relocation of a shard copy to a different node in the same cluster (`peer`).
+* A <<snapshots-restore-snapshot,snapshot restore>> operation (`snapshot`).
+* A <<indices-clone-index,clone>>, <<indices-shrink-index,shrink>>, or
+<<indices-split-index,split>> operation (`local_shards`).
```

Comment:
@DaveCTurner based on your feedback [here](https://github.com/elastic/elasticsearch/pull/109943#discussion_r1682710179) I've referenced the recovery source. can you verify that I've added the right type to the explanation? 

I'm also a little concerned that this will be confusing because I don't think we distinguish between `empty_store` and `existing_store` in the cat recovery API (the type is sometimes returned as just `store` [according to the API docs](https://www.elastic.co/guide/en/elasticsearch/reference/current/cat-recovery.html#cat-recovery-api-ex-dead)).

Reply:
I think the docs are wrong in that case, we do distinguish `empty_store` from `existing_store` in practice:

```
$ esget '_cat/recovery?h=type' | sort | uniq -c
   3 empty_store
   2 existing_store
  59 peer
```

New categories look good to me, tho I think I would not include the `quoted_names` of these things here. Either that or we should point out that those names appear in the `GET _recovery` and `GET _cat/recovery` APIs, and they're capitalized in the former but not in the latter.